models/Import/Service/ResultImporter.php

```diff
@@ -0,0 +1,369 @@
+    {
+        /** @var taoResultServer_models_classes_ResponseVariable $scoreTotalVariable */
+        $scoreTotalVariable = null;
+        $scoreTotal = null;
+        $scoreTotalMax = null;
+        $updatedScoreTotal = null;
+        $scoreTotalVariableId = null;
+        $outcomeVariables = $resultStorage->getDeliveryVariables($deliveryExecutionUri);
+
+        foreach ($outcomeVariables as $id => $outcomeVariable) {
+            $variable = $this->getVariable($outcomeVariable);
+
+            if ($variable === null) {
+                continue;
+            }
+
+            if ($variable->getIdentifier() === 'SCORE_TOTAL') {
+                $scoreTotalVariableId = $id;
+                $scoreTotal = (float)$variable->getValue();
+
+                $updatedScoreTotal = $scoreTotal;
+                $scoreTotalVariable = $variable;
+
+                continue;
+            }
+
+            if ($variable->getIdentifier() === 'SCORE_TOTAL_MAX') {
+                $scoreTotalMax = (float)$variable->getValue();
+            }
+        }
+
+        if ($scoreTotal === null) {
+            throw new ImportResultException(
+                sprintf(
+                    'SCORE_TOTAL is null for delivery execution %s',
+                    $deliveryExecutionUri
+                )
+            );
+        }
+
+        return [
+            'scoreTotalVariableId' => $scoreTotalVariableId,
+            'scoreTotalVariable' => $scoreTotalVariable,
+            'updatedScoreTotal' => $updatedScoreTotal,
+            'scoreTotalMax' => $scoreTotalMax,
+        ];
+    }
+
+    /**
+     * @throws ImportResultException
+     */
+    private function getItemVariable(
+        AbstractRdsResultStorage $resultStorage,
+        string $deliveryExecutionUri,
+        string $itemId,
+        string $callItemId,
+        string $variableIdentifier
+    ): array {
+        $variableVersions = $resultStorage->getVariable($callItemId, $variableIdentifier);
+        $lastVariable = is_array($variableVersions) ? (array)end($variableVersions) : [];
+
+        if (empty($variableVersions)) {
+            throw new ImportResultException(
+                sprintf(
+                    'Variable %s not found for item %s on delivery execution %s',
+                    $variableIdentifier,
+                    $itemId,
+                    $deliveryExecutionUri
+                )
+            );
+        }
+
+        /** @var taoResultServer_models_classes_Variable $variable */
+        $variable = $lastVariable['variable'] ?? null;
+
+        if (!$variable instanceof taoResultServer_models_classes_Variable) {
+            throw new ImportResultException(
+                sprintf(
+                    'Variable %s is typeof %s, expected instance of %s, for item %s and execution %s',
+                    $variableIdentifier,
+                    get_class($variable),
```

Comment:
We cannot use `get_class` on null.
In PHP 7.4 you will get a warning, but starting from PHP 8 there will be a fatal error
![image](https://user-images.githubusercontent.com/59471572/226357138-299d9f9e-db53-4c8f-abb8-2f72d8229911.png)

Reply:
Done!